pkg/internal/vault/vault.go

```diff
@@ -36,15 +36,24 @@ import (
 
 var _ Interface = &Vault{}
 
-type VaultClientBuilder func(namespace string, secretsLister corelisters.SecretLister,
+// ClientBuilder is a function type that returns a new Interface.
+// Can be used in tests to create a mock signer of Vault certificate requests.
+type ClientBuilder func(namespace string, secretsLister corelisters.SecretLister,
 	issuer v1.GenericIssuer) (Interface, error)
 
+// Interface implements various high level functionality related to connecting
+// with a Vault server, verifying its status and signing certificate request for
+// Vault's certificate.
+// TODO: do we need the Sys() *vault.Sys func given that
+// that implementation of this interface (Vault) also has a field (client) that
+// is an interface that also implements the same Sys() *vault.Sys function?
```

Comment:
I don't  understand that comment.

Reply:
👍🏼 Tried to improve it -> `TODO: Sys() is duplicated here and in Client interface`